### PR TITLE
Dbeaver/pro#2529 Clean up code after initial semantics-based completion implementation

### PR DESCRIPTION
**Pull request by E1izabeth**
Only code style fixes, java docs and moving some nested classes to a separate file

### CODE REVIEW

File: plugins/org.jkiss.dbeaver.model.lsm/src/org/jkiss/dbeaver/model/stm/STMSource.java [hunk 1]
```diff
@@ -23,16 +23,27 @@
...
 import java.io.Reader;
 
-
+/**
+ * Source stream for syntax analysis
+ */
 public interface STMSource {
 
+    /**
+     * Get characters strream
```
Comment (LonwoLonwo) [inline plugins/org.jkiss.dbeaver.model.lsm/src/org/jkiss/dbeaver/model/stm/STMSource.java]:
```suggestion
     * Get characters stream
```

File: plugins/org.jkiss.dbeaver.model.lsm/src/org/jkiss/dbeaver/model/stm/STMTreeNode.java [hunk 2]
```diff
@@ -7,22 +7,41 @@
...
+
+    /**
+     * Get the state of the antlr parser finite state machine associated with the entry point to the corresponding syntax rule for the text
+     */
     int getAtnState();
-    
+
+
+    /**
+     * Get the name of the syntaxt tree node
```
Comment (LonwoLonwo) [inline plugins/org.jkiss.dbeaver.model.lsm/src/org/jkiss/dbeaver/model/stm/STMTreeNode.java]:
```suggestion
     * Get the name of the syntax tree node
```

File: plugins/org.jkiss.dbeaver.model.lsm/src/org/jkiss/dbeaver/model/stm/STMUtils.java [hunk 3]
```diff
@@ -92,4 +90,66 @@ public static List<String> expandTermStrings(@NotNull STMTreeNode root) {
     }
 
 
+    public static <T, K> int binarySearchByKey(
```
Comment (LonwoLonwo) [inline plugins/org.jkiss.dbeaver.model.lsm/src/org/jkiss/dbeaver/model/stm/STMUtils.java:104]: Javadocs :(

Comment (E1izabeth) [reply plugins/org.jkiss.dbeaver.model.lsm/src/org/jkiss/dbeaver/model/stm/STMUtils.java:104]: I don't think, it's needed here. The name of the method is self-descriptive

File: plugins/org.jkiss.dbeaver.model.sql/src/org/jkiss/dbeaver/model/sql/semantics/SQLQueryQualifiedName.java [hunk 4]
```diff
@@ -1,171 +1,196 @@
...
+        if (this.schemaName != null) {
+            this.schemaName.getSymbol().setSymbolClass(symbolClass);
+        }
+        if (this.catalogName != null) {
+            this.catalogName.getSymbol().setSymbolClass(symbolClass);
+        }
+    }
+
+    /**
+     * Set the definition to the qaulfied name components based on the database metadata
```
Comment (LonwoLonwo) [inline plugins/org.jkiss.dbeaver.model.sql/src/org/jkiss/dbeaver/model/sql/semantics/SQLQueryQualifiedName.java]:
```suggestion
     * Set the definition to the qualified name components based on the database metadata
```

File: plugins/org.jkiss.dbeaver.model.sql/src/org/jkiss/dbeaver/model/sql/semantics/SQLQueryQualifiedName.java [hunk 5]
```diff
@@ -1,171 +1,196 @@
...
+                    } else {
+                        this.catalogName.getSymbol().setSymbolClass(SQLQuerySymbolClass.CATALOG);
+                    }
+                }
+            }
+        }
+    }
+
+    /**
+     * Set the definition to the qaulfied name components based on the query structure
```
Comment (LonwoLonwo) [inline plugins/org.jkiss.dbeaver.model.sql/src/org/jkiss/dbeaver/model/sql/semantics/SQLQueryQualifiedName.java]:
```suggestion
     * Set the definition to the qualified name components based on the query structure
```

**Pull request merged**